### PR TITLE
Taking advantage of data_buffer in the DEM elements

### PR DESCRIPTION
**Pull request by GuillermoCasas**
This branch passes all the DEM tests.

### CODE REVIEW

**Review by joaquinirazabal — COMMENTED** [review]
The Rolling Friction part looks good for me! Good job, Guillermo ;)

File: applications/DEM_application/custom_elements/spheric_particle.cpp [hunk 1]
```diff
@@ -146,163 +146,73 @@ void SphericParticle::SetIntegrationScheme(DEMIntegrationScheme::Pointer& integr
...
-    array_1d<double, 3> additional_forces;
-    array_1d<double, 3> additionally_applied_moment;
-    array_1d<double, 3>& elastic_force       = this->GetGeometry()[0].FastGetSolutionStepValue(ELASTIC_FORCES);
-    array_1d<double, 3>& contact_force       = this->GetGeometry()[0].FastGetSolutionStepValue(CONTACT_FORCES);
-    array_1d<double, 3>& rigid_element_force = this->GetGeometry()[0].FastGetSolutionStepValue(RIGID_ELEMENT_FORCE);
+
+    // Creating a data buffer to store those variables that we want to reuse so that we can keep function parameter lists short
+
+    std::unique_ptr<ParticleDataBuffer> p_buffer = CreateParticleDataBuffer(this); // all memory will be freed once this shared pointer goes out of scope
+    ParticleDataBuffer& data_buffer = *p_buffer;
```
Comment (maceligueta) [inline applications/DEM_application/custom_elements/spheric_particle.cpp:153]: +1 for the unique_ptr here!

**Review by maceligueta — CHANGES_REQUESTED** [review]
Nice implementation, we will cean up some arguments with more time. However, do you need to upload all those txt files? There's also an h5py file and a couple of files called 'TimesPartialRelease'. If you need them for the tests ok.

**Comment by GuillermoCasas** [comment]
No, those slipped in the commit by error

**Review by maceligueta — APPROVED** [review]

**Pull request merged**